### PR TITLE
Update timestamp for adding and removing grouping admin

### PR DESCRIPTION
**Pull request by hokwaichan**
# Ticket Link

[Groupings-1542](https://uhawaii.atlassian.net/browse/GROUPINGS-1542)

# List of squashed commits
 - Fixed remove admin timestamp issue  uh-settings:groupingsAdmins
 - Fixed add admin timestamp issue uh-settings:groupingsAdmins
 - Fully covered unit test

# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Integration Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

### CODE REVIEW

**Review by mhodgesatuh — APPROVED** [review]
Looks good as far as I can tell.  Glad to see the method names becoming more specific.

I didn't see timestamp tests, but recall these might be done later and for now the point is to get prod (devrc-2-4) fixed ASAP.

File: src/test/java/edu/hawaii/its/api/service/TestGroupingAssignmentService.java
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/service/TestGroupingAssignmentService.java:1]: Why was this file deleted?

File: src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java [hunk 1]
```diff
@@ -13,12 +13,7 @@
 import edu.hawaii.its.api.wrapper.RemoveMembersResults;
 import edu.hawaii.its.api.wrapper.SubjectsResults;
 
-import edu.internet2.middleware.grouperClient.ws.beans.WsAddMemberResults;
-import edu.internet2.middleware.grouperClient.ws.beans.WsDeleteMemberResult;
-import edu.internet2.middleware.grouperClient.ws.beans.WsDeleteMemberResults;
-import edu.internet2.middleware.grouperClient.ws.beans.WsFindGroupsResults;
-import edu.internet2.middleware.grouperClient.ws.beans.WsGetSubjectsResults;
-import edu.internet2.middleware.grouperClient.ws.beans.WsHasMemberResults;
+import edu.internet2.middleware.grouperClient.ws.beans.*;
```
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java:16]: Could you undo this * import? Thanks.

File: src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java [hunk 2]
```diff
@@ -433,4 +497,55 @@ public void optOutTest() {
...
+        subjectsResults = new SubjectsResults(wsGetSubjectsResults);
+        assertNotNull(subjectsResults);
+        doReturn(subjectsResults).when(grouperApiService).getSubjects("bogus-identifier");
+
+        assertThrows(UhMemberNotFoundException.class,
+                () -> updateMemberService.removeAdminMember(TEST_USERNAMES.get(1), "bogus-identifier"));
+    }
+
+
+ */
```
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java:536]: I believe we created this test but we were not sure if it should be included in this PR. Let's uncomment this and include it.

File: src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java:1]: Could you move the private methods addAdmin and removeAdmin to the bottom of the class where the other private methods are placed? Thanks.

File: src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java (hunk 2 above)
Comment (hokwaichan) [reply src/test/java/edu/hawaii/its/api/service/UpdateMemberServiceTest.java:536]: I will remove it since it is an old version I used for testing

File: src/main/java/edu/hawaii/its/api/service/GrouperApiService.java [hunk 3]
```diff
@@ -282,7 +282,8 @@ public RemoveMemberResult removeMember(String currentUser, String groupPath, Str
         return exec.execute(new RemoveMembersCommand()
                 .owner(currentUser)
                 .assignGroupPath(groupPath)
-                .addUhIdentifier(uhIdentifier)).getResults().get(0);
+                .addUhIdentifier(uhIdentifier))
+                .getResults().get(0);
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/service/GrouperApiService.java:286]: Could you move this line back onto the previous line? Thanks.

File: src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java [hunk 4]
```diff
@@ -40,6 +40,7 @@ public class UpdateMemberService {
     @Autowired
     private UpdateTimestampService timestampService;
 
+
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/service/UpdateMemberService.java:43]: Extra space.

File: src/test/java/edu/hawaii/its/api/service/TestGroupingAssignmentService.java [hunk 5]
```diff
@@ -199,4 +197,4 @@ public void isSoleOwner() {
         updateMemberService.removeOwnership(ADMIN, GROUPING, testUid);
     }
 
-}
+}
```
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/service/TestGroupingAssignmentService.java:200]: End of line error here.

**Review by JorWo — APPROVED** [review]
Looks great! Thanks for adding the integration test for UpdateTimestampService.
